src/treelearner/feature_histogram.hpp

```diff
@@ -595,7 +595,7 @@ class FeatureHistogram {
 
         double sum_left_gradient = sum_gradient - sum_right_gradient;
         if (IS_RAND) {
-          if (t + offset != rand_threshold) {
+          if (t - 1 + offset != rand_threshold) {
```

Comment:
is there a unit test we can write that would have caught this and would prevent it from being re-introduced? Or is this too deep in the code to target with a test?

Reply:
I think we should enable c++ test, but I don't have effort for it..